### DISCUSSION TITLE
type="module" in externalResources

### DISCUSSION THREAD

**Discussion by markacianfrani** [codesandbox/sandpack#998] [Ideas]
Would it be possible to allow <script type="module"...> in externalResources so that I could use ES modules? Right now it just accepts an array of strings and just checks if css or js. I was thinking maybe somewhere [here](https://github.com/codesandbox/sandpack/blob/main/sandpack-client/src/clients/static/index.ts#L129) could check if the resources ends in '.esm.js'. Not everyone may follow that naming convention though. 

My use case is wanting to play around with our web component library, which is hosted on a CDN as an ES Module, but I can't load it via externalResources because I can't specify `type="module"`.

**Comment by jon-schmedt-alight** [comment]
This would be really cool!  One use case that I can think of would be to use a loader and defineCustomElements as well.
